odoo/addons/test_new_api/tests/test_new_fields.py

```diff
@@ -1657,11 +1659,29 @@ def test_field(field_name, truthy_values, operations):
             test_cases(field_name, operations, truthy_values[0])
 
         def test_cases(field_name, operations, default=None):
+            model = self.env['test_new_api.company']
+            field = model._fields[field_name]
+            field_fallback = field.get_company_dependent_fallback(model)
+            record_fallback[field_name] = field_fallback
+            current_thread = threading.current_thread()
+
             for operator, values in operations.items():
                 for value in values:
                     domain = [(field_name, operator, value)]
+                    company_dependent_column_not_null = not record_fallback.filtered_domain(domain)
                     with self.subTest(domain=domain, default=default):
-                        search_result = Model.search([('id', 'in', records.ids)] + domain)
+                        if company_dependent_column_not_null:
+                            Model.search([('id', 'in', records.ids)] + domain)
+                            current_thread.query_count = 0
+                            current_thread.query_time = 0
+                            search_result = Model.search([('id', 'in', records.ids)] + domain)
+                            if current_thread.query_count:
+                                # parent_of and child_of may need extra queries
+                                expected_contained_sqls = [''] * (current_thread.query_count - 1) + [f'"test_new_api_company"."{field_name}" IS NOT NULL']
+                                with self.assertQueries(expected_contained_sqls, operator='contain'):
+                                    Model.search([('id', 'in', records.ids)] + domain)
```

Comment:
What's the deal with the current thread?  The query count should be predictable, shouldn't it?

Reply:
There might be multiple queries, all queries except the last one are used to check if the fallback value can be ignored in the search. Only the last query is the one we test to check if `col IS NOT NULL` is containted.
for example 
if the `fallback_res_company` is a res.company record for the field's fallback
('company_id', 'ilike', 'xxx') need to firstly call some extra queries to fetch and compute the `fallback_res_company.display_name` to know if the result `ilike` `xxx` or not